src/main/scala/io/conduktor/ksm/source/HttpSourceAcl.scala

```diff
@@ -0,0 +1,198 @@
+    *
+    * @return
+    */
+  override def refresh(): Option[ParsingContext] = {
+
+    val request: Request = new Request(uri)
+    // super important in order to properly fail in case a timeout happens for example
+    request.enableThrowingIOException(true)
+    request.followRedirects(false)
+    request.connectTimeoutMillis(Int.MaxValue)
+    request.readTimeoutMillis(Int.MaxValue)
+
+    if (enableAuth) {
+      val authorization = getIdToken()
+      request.header("Authorization", authorization)
+    }
+
+    // we use this header for the 304
+    lastModified.foreach(header => request.header("If-Modified-Since", header))
+    val response: Response = HTTP.get(request)
+
+    response.status match {
+      case 200 =>
+        lastModified = response.header("Last-Modified")
+
+        Some(
+          ParsingContext(
+            parserRegistry.getParser(this.parser),
+            new BufferedReader(new StringReader(response.textBody))
+          )
+        )
+      case 304 =>
+        None
+      case _ =>
+        // we got an http error so we propagate it
+        log.error("HTTP Status {} in HTTP Source request: {}", response.status, response.asString)
+        throw HTTPException(Some(response.asString), response)
+    }
+  }
+
+  private def getIdToken(): String = {
+    val audience = this.targetAudience
+
+    if (jwtMap.get(audience) != null
+      && expirationMap.get(audience) != null
+      && Instant
+      .now()
+      .isBefore(expirationMap.get(audience).getOrElse(Instant.MIN)))
+      return jwtMap.get(audience).get
+
+    log.info("Getting IdToken from Google Cloud")
+
+    try {
+      val jwt = this.getJwtFromGoogleIam
+      // Add cache for 30 minutes
+      log.info("Adding Token to cache for 30 minutes")
+      jwtMap.put(audience, jwt)
+      expirationMap.put(audience, Instant.now().plus(30, ChronoUnit.MINUTES))
+      jwt
+    } catch {
+      case e: IOException =>
+        throw new RuntimeException("Couldn't get IdToken", e)
+    }
+  }
+
+  import java.io.IOException
+
+  @throws[IOException]
+  private def getJwtFromGoogleIam: String = {
+    val credentials = if (serviceAccountKey == null || serviceAccountKey.isEmpty) {
+      log.info("Getting Google Default Application Credentials...")
+      GoogleCredentials.getApplicationDefault
+    } else {
+      log.info("Getting Google Credentials from Key File...")
+      GoogleCredentials
+        .fromStream(new ByteArrayInputStream(this.serviceAccountKey.getBytes))
+        .createScoped(Collections.singleton(IAM_SCOPE))
+    }
+
+    if (!credentials.isInstanceOf[IdTokenProvider])
+      throw new RuntimeException(
+        s"Google Credentials: wrong type of token provider. Expected: IdTokenProvider, got: $credentials"
+      )
+
+    val idTokenCredentials = IdTokenCredentials.newBuilder
+      .setIdTokenProvider(credentials.asInstanceOf[ServiceAccountCredentials])
```

Comment:
please use a `match` instead of `isInstanceOf` and check directly for the needed type (`ServiceAccountCredentials`)

Reply:
Done!